src/dmd/typesem.d

```diff
@@ -1472,6 +1474,10 @@ extern(C++) Type typeSemantic(Type t, Loc loc, Scope* sc)
                         fparam.storageClass &= ~STC.auto_;    // https://issues.dlang.org/show_bug.cgi?id=14656
                         fparam.storageClass |= STC.autoref;
                     }
+                    else if (mtype.incomplete)
+                    {
+                        // the default argument may have been temporarily removed
```

Comment:
I'd link to the bugzilla issue here, that is a comment that would really get its message across better with more context.